src/mlpack/core/data/image_resize_crop.hpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file core/data/image_resize_crop.hpp
+ * @author Omar Shrit
+ *
+ * Image resize and crop functionalities.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+#define MLPACK_CORE_DATA_IMAGE_RESIZE_CROP_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+
+
+void resize(const unsigned char* image_data, int s_width, int s_height,
+    unsigned char*& frame_buffer_out, int d_width, int d_height, int num_channel)
+{
+  stbir_resize_uint8(image_data, s_width, s_height, 0,
+                     frame_buffer_out, d_width, d_height, 0, num_channel);
+}
+
+
+void resize_and_crop(const unsigned char* image_data, int s_width, int s_height,
+    unsigned char*& frame_buffer_out, int d_width, int d_height, int num_channel)
```

Comment:
I'm not sure I follow this one---is your goal just to crop, or to first resize and then crop?  I would suggest a fairly similar API to the one above.

I also wonder if it would be cleaner to have a `CropImage()` standalone function, and a `ResizeImage()` function too, and then users could just call both in sequence to imitate the support of this function (but like I said I am not 100% sure what the intention here is).